lib/matplotlib/tests/test_figure.py

```diff
@@ -533,3 +533,202 @@ def test_removed_axis():
     fig, axs = plt.subplots(2, sharex=True)
     axs[0].remove()
     fig.canvas.draw()
+
+
+class TestCollage:
+    @check_figures_equal(extensions=['png'])
+    @pytest.mark.parametrize('x', [
+        [["A", "A", "B"], ["C", "D", "B"]],
+        [[1, 1, 2], [3, 4, 2]]]
+    )
+    def test_basic(self, fig_test, fig_ref, x):
+        grid_axes = fig_test.build_grid(x)
+
+        for k, ax in grid_axes.items():
+            ax.set_title(k)
+
+        labels = sorted(np.unique(x))
+
+        assert len(labels) == len(grid_axes)
+
+        gs = fig_ref.add_gridspec(2, 3)
+        axA = fig_ref.add_subplot(gs[:1, :2])
+        axA.set_title(labels[0])
+
+        axB = fig_ref.add_subplot(gs[:, 2])
+        axB.set_title(labels[1])
+
+        axC = fig_ref.add_subplot(gs[1, 0])
+        axC.set_title(labels[2])
+
+        axD = fig_ref.add_subplot(gs[1, 1])
+        axD.set_title(labels[3])
+
+    @check_figures_equal(tol=0.005, extensions=['png'])
```

Comment:
why do any of these need nonzero tol?

Reply:
The source of this was that there was some overlap between the text but apparently `build_grid` and the test add the axes in the opposite order so there was some anti-aliasing layering noise.  Fixed the tests to de-conflict the text and everything is passing with 0 tolerance.